src/control/server/config/server.go

```diff
@@ -700,8 +707,12 @@ func (cfg *Server) Validate(log logging.Logger) (err error) {
 		return FaultConfigNoProvider
 	case cfg.ControlPort <= 0:
 		return FaultConfigBadControlPort
-	case cfg.TelemetryPort < 0:
-		return FaultConfigBadTelemetryPort
+	}
+
+	if cfg.TelemetryConfig != nil {
+		if cfg.TelemetryConfig.Port < 0 {
+			return FaultConfigBadTelemetryPort
+		}
```

Comment:
Anything else we want to verify here? Telemetry_config adds more parameters.

Reply:
I checked and looks like we may not need to check anything else it as other parameters are getting verified in multiple places. Some of the parameter are bool (enabled/allowinsecure)  so if it provided some thing else it will fail during assignments

```
 yaml: unmarshal errors:
  line 34: cannot unmarshal !!str `theydhd` into bool
```

Some of the string parameter is also checked so not sure we need this check explicitly. 
`if cfg.TelemetryConfig.HttpsCert == "" || cfg.TelemetryConfig.HttpsKey == ""`

Reply:
OK. Just brought it up in case there was something more interesting to check.